hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -202,22 +205,26 @@ public ContainerInfo allocateContainer(
             " matching pipeline for replicationConfig: " + replicationConfig
             + ", State:PipelineState.OPEN", e);
       }
-      pipelineManager.acquireReadLock();
-      lock.lock();
+
       try {
-        pipelines = pipelineManager
-            .getPipelines(replicationConfig, Pipeline.PipelineState.OPEN);
-        if (!pipelines.isEmpty()) {
-          pipeline = pipelines.get(random.nextInt(pipelines.size()));
-          containerInfo = createContainer(pipeline, owner);
-        } else {
-          throw new IOException("Could not allocate container. Cannot get any" +
-              " matching pipeline for replicationConfig: " + replicationConfig
-              + ", State:PipelineState.OPEN");
+        lock.lock();
+        try {
+          pipelineManager.acquireReadLock();
```

Comment:
Is there any reason for acquiring locks inside try blocks? If not, it's better to move them outside the try blocks. 

My reasoning: If some implementation of these locks ends up throwing an unchecked exception when trying to lock, then we won't acquire the lock but still try to unlock it in finally. That will throw a runtime exception.

Reply:
I was trying to make findbugs happy, as before I had the two nested try blocks, findbugs complained. I'll see if I can get it happy while having the locks outside of the try block.